KutokAccounting.Services/TransactionTypes/TransactionTypeRepository.cs

```diff
@@ -0,0 +1,145 @@
+using KutokAccounting.DataProvider;
+using KutokAccounting.DataProvider.Models;
+using KutokAccounting.Services.TransactionTypes.Interfaces;
+using KutokAccounting.Services.TransactionTypes.Models;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace KutokAccounting.Services.TransactionTypes;
+
+public class TransactionTypeRepository : ITransactionTypeRepository
+{
+    private readonly KutokDbContext _dbContext;
+    private readonly SemaphoreSlim _semaphoreSlim;
+    private readonly ILogger<TransactionTypeRepository> _logger;
+    
+    public TransactionTypeRepository(KutokDbContext dbContext, [FromKeyedServices(KutokConfigurations.WriteOperationsSemaphore)] SemaphoreSlim semaphoreSlim, ILogger<TransactionTypeRepository> logger)
+    {
+        _dbContext = dbContext;
+        _logger = logger;
+        _semaphoreSlim = semaphoreSlim;
+    }
+    public async ValueTask CreateAsync(TransactionType transactionType, CancellationToken cancellationToken)
+    {
+        await _semaphoreSlim.WaitAsync(cancellationToken);
+
+        try
+        {
+            await _dbContext.TransactionTypes.AddAsync(transactionType, cancellationToken);
+            await _dbContext.SaveChangesAsync(cancellationToken);
+        }
+        catch (Exception e)
+        {
+            _logger.LogInformation(e, "Failed to create transaction type with Name: {TransactionTypeName}", transactionType.Name);
+        }
+        finally
+        {
+            _semaphoreSlim.Release();
+        }
+    }
```

Comment:
Коли більше двох параметрів, спускай їх по поверхах. І відступ між методами в один рядок, будь-ласка
```suggestion
    public TransactionTypeRepository(KutokDbContext dbContext, 
    [FromKeyedServices(KutokConfigurations.WriteOperationsSemaphore)] SemaphoreSlim semaphoreSlim, 
    ILogger<TransactionTypeRepository> logger)
    {
        _dbContext = dbContext;
        _logger = logger;
        _semaphoreSlim = semaphoreSlim;
    }
    
    public async ValueTask CreateAsync(TransactionType transactionType, CancellationToken cancellationToken)
    {
        await _semaphoreSlim.WaitAsync(cancellationToken);

        try
        {
            await _dbContext.TransactionTypes.AddAsync(transactionType, cancellationToken);
            await _dbContext.SaveChangesAsync(cancellationToken);
        }
        catch (Exception e)
        {
            _logger.LogInformation(e, "Failed to create transaction type with Name: {TransactionTypeName}", transactionType.Name);
        }
        finally
        {
            _semaphoreSlim.Release();
        }
    }
```